tests/NLog.UnitTests/LogMessageFormatterTests.cs

```diff
@@ -34,21 +34,22 @@
 namespace NLog.UnitTests
 {
     using Xunit;
+    using System.Collections.Generic;
 
     public class LogMessageFormatterTests : NLogTestBase
     {
         [Fact]
         public void StructuredLoggingTest()
         {
             LogEventInfo logEventInfo = new LogEventInfo(LogLevel.Info, string.Empty, "Login request from {Username} for {Application}");
-            logEventInfo.Properties["Username"] = "John";
-            logEventInfo.Properties["Application"] = "BestApplicationEver";
-            logEventInfo.Parameters = new object[] { "Login request from John for BestApplicationEver" };   // Must have parameter to activate MessageFormatter
+            foreach (var item in new KeyValuePair<string, object>[] { new KeyValuePair<string, object>("Username", "John"), new KeyValuePair<string, object>("Application", "BestApplicationEver") })
+                logEventInfo.Properties.AddOrderedProperty(item.Key, item.Value);
+            logEventInfo.Parameters = new object[] { "John", "BestApplicationEver", "Login request from John for BestApplicationEver" };
```

Comment:
Why do we need this beside the `logEventInfo.Properties`? I think this unit test is doing to much, as I'm confused. Are we testing multiple things here?

Reply:
For normal structured logging you get a raw-message and arguments-params. Then based on that you derive the FormattedMessage and fill Properties-dictionary. In the Microsoft Extension Logging world you get a pre-formatted-message and can dig out some message-properties (where the last property is actually the original raw-message). For the Microsoft Extension Logging, then I want to capture the raw-message, capture the microsoft-formatted-message (store it as arguments-params), and fill the Properties-dictionary.